src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,103 @@
-import { useState } from 'react';
+import React, { FormEvent, useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void,
+};
+
+const initialFormState: Movie = {
+  title: '',
+  description: '',
+  imgUrl: '',
+  imdbUrl: '',
+  imdbId: '',
+};
+
+export const NewMovie: React.FC<Props> = (props) => {
+  const {
+    onAdd,
+  } = props;
+  const [count, setCount] = useState(0);
+  const [formState, setFormState] = useState(initialFormState);
+
+  const {
+    title,
+    description,
+    imgUrl,
+    imdbUrl,
+    imdbId,
+  } = formState;
+
+  const allRequiredFieldsAreFilled = formState.title
+  && formState.imgUrl && formState.imdbUrl && formState.imdbId;
```

Comment:
here is my suggestion to prevent this behavior
![image](https://github.com/mate-academy/react_movies-list-add-form/assets/96138503/d5809c46-38f0-4cbf-aa1f-65941eeb7b18)

```suggestion
  const allRequiredFieldsAreFilled = title.trim() && imgUrl.trim() && imdbUrl.trim() && imdbId.trim();
```